Move math docs out of HelpDB, more examples, fix typos

kshyatt opened the pull request:
Found a series of typos in `cov` and friends. Added more notes
about `NaN` and Julia. Made the function signatures reflect
what's actually in the code. More examples for quite a few
functions.

kshyatt committed:
Move math docs out of HelpDB, more examples, fix typos

Found a series of typos in `cov` and friends. Added more notes
about `NaN` and Julia. Made the function signatures reflect
what's actually in the code. More examples for quite a few
functions.

kshyatt committed:
Move quadgk docs out, update formatting

kshyatt committed:
Moved special functions out of HelpDB, insert some links

kshyatt committed:
Updated docs for some array ops as well

tkelman commented on base/array.jl:
````diff
@@ -658,6 +658,40 @@ hvcat(rows::Tuple{Vararg{Int}}, xs::Union{Vector,Matrix}...) = typed_hvcat(promo
+ 0.682085  0.179069   0.198193
+```
+"""
 function hcat{T}(V::Vector{T}...)
````
this is only for vectors, the docstring is more general

tkelman commented on base/statistics.jl:
```diff
@@ -493,6 +569,17 @@ end
+
+Compute the median of whole array `v`, or optionally along the dimensions in `region`. For
+even number of elements no exact median element exists, so the result is equivalent to
+calculating mean of two median elements.
```
some missing articles - "Compute the median of the whole array `v`, or optionally along the dimensions in `region`. For an even number of elements no exact median element exists, so the result is equivalent to calculating the mean of two median elements."

maybe a "the" before the "two" as well?

tkelman commented on base/rational.jl:
```diff
@@ -90,8 +90,27 @@ promote_rule{T<:Integer,S<:AbstractFloat}(::Type{Rational{T}}, ::Type{S}) = prom
 widen{T}(::Type{Rational{T}}) = Rational{widen(T)}
 
+"""
+    rationalize([Type<:Integer,] x; tol::Real=eps(x))
```
I think `T<:Integer` might be more consistent. And should mention that if not provided, `T` defaults to `Int`.

tkelman commented on base/array.jl:
````diff
@@ -667,6 +701,28 @@ function hcat{T}(V::Vector{T}...)
+ 0.574442  0.525314  0.3796    0.486924  0.572558
+```
+"""
 function vcat{T}(arrays::Vector{T}...)
````
likewise this signature is only for vectors, docstring is more general

tkelman commented on base/special/bessel.jl:
```diff
@@ -51,17 +56,53 @@ function airy(k::Integer, z::Complex128)
+for `k == 0 || k == 1`, and ``\\operatorname{Ai}(x) e^{- \\left| \\operatorname{Re} \\left( \\frac{2}{3} x \\sqrt{x} \\right) \\right|}``
+for `k == 2 || k == 3`.
+"""
 function airyx(k::Integer, z::Complex128)
```
this method is for more specific types than the docstring

tkelman commented on base/abstractarraymath.jl:
```diff
@@ -11,13 +11,39 @@ transpose(a::AbstractArray) = error("transpose not implemented for $(typeof(a)).
+"""
+    vec(a::AbstractArray) -> Vector
+
+Vectorize an array using column-major convention.
```
Worth rephrasing this, not immediately obvious what this means.

tkelman commented on base/math.jl:
````diff
@@ -43,6 +43,21 @@ clamp{X,L,H}(x::X, lo::L, hi::H) =
+ 9.000000000000000000000000000000000000000000000000000000000000000000000000000000
+```
+"""
 clamp{T}(x::AbstractArray{T}, lo, hi) =
````
the docstring is more general, there are also methods for scalar input

tkelman commented on doc/stdlib/arrays.rst:
```diff
@@ -437,12 +437,58 @@ Indexing, Assignment, and Concatenation
 
+   .. doctest::
+
+       julia> a = rand(1,5)
```
`rand` isn't a good thing to put in a doctest unless you `srand`

tkelman commented on base/abstractarraymath.jl:
```diff
@@ -11,13 +11,39 @@ transpose(a::AbstractArray) = error("transpose not implemented for $(typeof(a)).
+
+Remove the dimensions specified by `dims` from array `A`.
+Elements of `dims` must be unique and within the range `1:ndims(A)`.
+`size(A,i)` must equal 1 for all `i` in `dims.
```
missing closing backtick after dims

tkelman commented on base/statistics.jl:
```diff
@@ -198,12 +220,41 @@ stdm(A::AbstractArray, m::Number; corrected::Bool=true) =
+Compute the sample standard deviation of a vector `v`
+with known mean `m`. If `corrected` is `true`,
+then the sum is scaled with `n-1` whereas the sum is
+scaled with `n` if `corrected` is false` where `n = length(x)`.
```
missing opening backtick before `false`

tkelman commented:
Should all the recommendations about DataArrays be changed to recommend NullableArrays now?

kshyatt commented:
>Should all the recommendations about DataArrays be changed to recommend NullableArrays now?

cc: @johnmyleswhite @davidagold

kshyatt committed:
Updated in response to feedback

Removed calls to `rand` in doctests. Made examples
better. Cleaned up function signatures.

johnmyleswhite commented:
Let's change that after DataFrames changes to use NullableArrays. As it stands, we're in a weird intermediate state in which there's no clear standard.

kshyatt committed:
Fix doctests

kshyatt committed:
Fix doctests ... again!
